backend/parser/src/main/java/me/codvis3D/ast/cpplisteners/CppLstnr_Initial.java

```diff
@@ -113,6 +178,135 @@ public void enterExpressionstatement(CPP14Parser.ExpressionstatementContext ctx)
 	    this.scopeStack.peek().addDataInModel(ctx.getText());
 	}
 
+	/**
+	 * Fetches a meta data of the function.
+	 *
+	 * @param      ctx   The context
+	 *
+	 * @return     return JSON object containing list of parameters, declrator ID, and scope function is from.
+	 */
+	private JSONObject fetchMetaData(CPP14Parser.FunctiondefinitionContext ctx){
+		JSONObject functionMetaData = new JSONObject();
+		CPP14Parser.PtrdeclaratorContext ptrdeclarator = null;
+	  	CPP14Parser.NoptrdeclaratorContext noptrdeclarator = null;
+
+	  	if (ctx.declarator().ptrdeclarator() != null){
+	  		ptrdeclarator = ctx.declarator().ptrdeclarator();
+	  	} else {
+	  		noptrdeclarator = ctx.declarator().noptrdeclarator();
+	  	}
+
```

Comment:
Could simplify this by making a listener that does nothing while it finds what you need and then overwriting that handler once you enterFunctionDefinition and exitFunctionDefinition. having a listener live only in that interval.
Would make it significantly easier to test (lower cyclometric complexity)